Assets/MixedRealityToolkit.Extensions/SceneTransitionService/ISceneTransitionService.cs

```diff
@@ -0,0 +1,133 @@
+        float TransitionProgress { get; }
+
+        /// <summary>
+        /// Fades out, enables progress indicator, execute scene operations in order, disables progress indicator, fades back in
+        /// </summary>
+        /// <param name="sceneOperations">A set of tasks from the Scene System.</param>
+        /// <param name="progressIndicator">If null, default progress indicator prefab will be used (or none if default is disabled in profile)</param>
+        /// <returns></returns>
+        Task DoSceneTransition(IEnumerable<Func<Task>> sceneOperations, IProgressIndicator progressIndicator = null);
+
+        /// <summary>
+        /// Fades out, enables progress indicator, executes scene op 1, executes scene op 2, disables progress indicator, fades back in
+        /// </summary>
+        /// <param name="sceneOp1"></param>
+        /// <param name="sceneOp2"></param>
+        /// <param name="progressIndicator"></param>
+        /// <returns></returns>
+        Task DoSceneTransition(Func<Task> sceneOp1, Func<Task> sceneOp2, IProgressIndicator progressIndicator = null);
+
+        /// <summary>
+        /// Fades out, enables progress indicator, execute scene operation, disables progress indicator, fades back in
+        /// </summary>
+        /// <param name="sceneOperations">A set of tasks from the Scene System.</param>
+        /// <param name="progressIndicator">If null, default progress indicator prefab will be used (or none if default is disabled in profile)</param>
+        /// <returns></returns>
+        Task DoSceneTransition(Func<Task> sceneOperation, IProgressIndicator progressIndicator = null);
+
+        /// <summary>
+        /// If FadeTargets is set to custom, you will need to provide a custom set of cameras for fading using this function PRIOR to calling DoSceneTransition.
+        /// </summary>
+        /// <param name="customFadeTargetCameras"></param>
+        void SetCustomFadeTargetCameras(IEnumerable<Camera> customFadeTargetCameras);
+
+        /// <summary>
+        /// Fades target cameras out to color. Can be used independently of scene transitions provided no transition is taking place.
+        /// </summary>
+        /// <param name="fadeColor"></param>
+        /// <param name="targetCameras"></param>
+        /// <returns></returns>
+        Task FadeOut();
+
+        /// <summary>
+        /// Fades target cameras in. Instant fade-out will occur if fade state is not opaque. Can be used independently of scene transitions provided no transition is taking place.
+        /// </summary>
+        /// <param name="fadeColor"></param>
+        /// <param name="targetCameras"></param>
+        /// <returns></returns>
+        Task FadeIn();
+
+        /// <summary>
+        /// Instantiates the default progress indicator and returns its main transform. Can be used independently of scene transitions provided no transition is taking place.
+        /// </summary>
+        /// <returns></returns>
```

Comment:
>/// <returns></returns> [](start = 8, length = 23)

Super nit: Be consistent with the usage of <return> - other places we omit it, and it's probably better to clean up all instances where it's empty because it doesn't add anything.